tests/evm_backends/pyevm_env.py

```diff
@@ -0,0 +1,222 @@
+
+    @property
+    def block_number(self) -> int:
+        return self._state.block_number
+
+    @property
+    def timestamp(self) -> int | None:
+        return self._state.timestamp
+
+    @property
+    def last_result(self) -> ExecutionResult:
+        result = self._last_computation
+        return ExecutionResult(
+            is_success=not result.is_error,
+            logs=list(_parse_log_entries(result)),
+            gas_refunded=result.get_gas_refund(),
+            gas_used=result.get_gas_used(),
+        )
+
+    def execute_code(
+        self,
+        to: str,
+        sender: str | None = None,
+        data: bytes | str = b"",
+        value: int = 0,
+        gas: int | None = None,
+        gas_price: int = 0,
+        is_modifying: bool = True,
+        blob_hashes: Optional[list[bytes]] = None,  # for blobbasefee >= Cancun
+    ):
+        data = data if isinstance(data, bytes) else bytes.fromhex(data.removeprefix("0x"))
+        sender = _addr(sender or self.deployer)
+        try:
+            computation = self._state.computation_class.apply_message(
+                state=self._state,
+                message=Message(
+                    to=_addr(to),
+                    sender=sender,
+                    data=data,
+                    code=self.get_code(to),
+                    value=value,
+                    gas=self.gas_limit if gas is None else gas,
+                    is_static=not is_modifying,
+                ),
+                transaction_context=BaseTransactionContext(origin=sender, gas_price=gas_price),
+            )
+        except VMError as e:
+            # py-evm raises when user is out-of-funds instead of returning a failed computation
+            raise EvmError(*e.args) from e
+
+        self._check_computation(computation)
+        return computation.output
+
+    def clear_transient_storage(self) -> None:
+        try:
+            self._state.clear_transient_storage()
+        except AttributeError as e:
+            assert e.args == ("No transient_storage has been set for this State",)
+
+    def _check_computation(self, computation):
+        self._last_computation = computation
+        if computation.is_error:
+            if isinstance(computation.error, Revert):
+                (output,) = computation.error.args
+                gas_used = computation.get_gas_used()
+                self._parse_revert(output, computation.error, gas_used)
+
+            raise EvmError(*computation.error.args) from computation.error
+
+    def get_code(self, address: str):
+        return self._state.get_code(_addr(address))
+
+    def time_travel(self, num_blocks=1) -> None:
+        """
+        Move the block number forward by `num_blocks` and the timestamp forward by `time_delta`.
+        """
+        self._context._block_number += num_blocks
+        self._context._timestamp += num_blocks
+
+    def get_excess_blob_gas(self) -> Optional[int]:
+        return self._context.excess_blob_gas
```

Comment:
that missing underscore definitely made me do a double take

Reply:
They have getters but no setters: https://github.com/ethereum/py-evm/blob/d8df507e42c885ddeca2b64bbb8f10705f075d3e/eth/vm/execution_context.py#L99